src/transformers/modeling_tf_utils.py

```diff
@@ -656,143 +770,177 @@ def resize_token_embeddings(self, new_num_tokens=None) -> tf.Variable:
         Return:
             :obj:`tf.Variable`: Pointer to the input tokens Embeddings Module of the model.
         """
-        model_embeds = self._resize_token_embeddings(new_num_tokens)
-        if new_num_tokens is None:
-            return model_embeds
+        if new_num_tokens is None or new_num_tokens == self.config.vocab_size:
+            return self._find_weights(self.get_input_embeddings())
 
-        return model_embeds
+        model_embeds = self._resize_token_embeddings(new_num_tokens)
 
-    def _resize_token_embeddings(self, new_num_tokens):
-        # get_input_embeddings and set_input_embeddings need to be implemented in base layer.
-        base_model = getattr(self, self.base_model_prefix, self)
-        old_embeddings = base_model.get_input_embeddings()
-        new_embeddings = self._get_resized_embeddings(old_embeddings, new_num_tokens)
-        base_model.set_input_embeddings(new_embeddings)
         # Update base model and current model config
         self.config.vocab_size = new_num_tokens
-        base_model.vocab_size = new_num_tokens
-        return base_model.get_input_embeddings()
-
-    def _get_word_embeddings(self, embeddings):
-        if hasattr(embeddings, "word_embeddings"):
-            # TFBertEmbeddings, TFAlbertEmbeddings, TFElectraEmbeddings
-            return embeddings.word_embeddings
-        elif hasattr(embeddings, "weight"):
-            # TFSharedEmbeddings
-            return embeddings.weight
+
+        return model_embeds
+
+    def _find_weights(self, embedding_layer):
```

Comment:
still my comment here on the function naming -> thin `_find_weights` is not ideal.

-> a) It's only used for word embeddings
-> b) I think it's not a "find" method but just a "get" method. We don't have any methods called "find" in Transformers as far as I know

-> maybe rename it to `_get_word_embedding_weight`?

Reply:
Ooops thanks for the notice!!

Reply:
Ok I changed the name everywhere!